pkg/acquisition/modules/kinesis/kinesis.go

```diff
@@ -0,0 +1,473 @@
+	"github.com/crowdsecurity/crowdsec/pkg/leakybucket"
+	"github.com/crowdsecurity/crowdsec/pkg/types"
+	"github.com/davecgh/go-spew/spew"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	log "github.com/sirupsen/logrus"
+	"gopkg.in/tomb.v2"
+	"gopkg.in/yaml.v2"
+)
+
+type KinesisConfiguration struct {
+	configuration.DataSourceCommonCfg `yaml:",inline"`
+	StreamName                        string  `yaml:"stream_name"`
+	StreamARN                         string  `yaml:"stream_arn"`
+	UseEnhancedFanOut                 bool    `yaml:"use_enhanced_fanout"` //Use RegisterStreamConsumer and SubscribeToShard instead of GetRecords
+	AwsProfile                        *string `yaml:"aws_profile"`
+	AwsRegion                         string  `yaml:"aws_region"`
+	AwsEndpoint                       string  `yaml:"aws_endpoint"`
+	ConsumerName                      string  `yaml:"consumer_name"`
+	FromSubscription                  bool    `yaml:"from_subscription"`
+}
+
+type KinesisSource struct {
+	Config          KinesisConfiguration
+	logger          *log.Entry
+	kClient         *kinesis.Kinesis
+	shardReaderTomb *tomb.Tomb
+}
+
+type CloudWatchSubscriptionRecord struct {
+	MessageType         string                           `json:"messageType"`
+	Owner               string                           `json:"owner"`
+	LogGroup            string                           `json:"logGroup"`
+	LogStream           string                           `json:"logStream"`
+	SubscriptionFilters []string                         `json:"subscriptionFilters"`
+	LogEvents           []CloudwatchSubscriptionLogEvent `json:"logEvents"`
+}
+
+type CloudwatchSubscriptionLogEvent struct {
+	ID        string `json:"id"`
+	Message   string `json:"message"`
+	Timestamp int64  `json:"timestamp"`
+}
+
+func (k *KinesisSource) newClient() error {
+	var sess *session.Session
+
+	if k.Config.AwsProfile != nil {
+		sess = session.Must(session.NewSessionWithOptions(session.Options{
+			SharedConfigState: session.SharedConfigEnable,
+			Profile:           *k.Config.AwsProfile,
+		}))
+	} else {
+		sess = session.Must(session.NewSessionWithOptions(session.Options{
+			SharedConfigState: session.SharedConfigEnable,
+		}))
+	}
+
+	if sess == nil {
+		return fmt.Errorf("failed to create aws session")
+	}
+	config := aws.NewConfig()
+	if k.Config.AwsRegion != "" {
+		config = config.WithRegion(k.Config.AwsRegion)
+	}
+	if k.Config.AwsEndpoint != "" {
+		config = config.WithEndpoint(k.Config.AwsEndpoint)
+	}
+	k.kClient = kinesis.New(sess, config)
+	if k.kClient == nil {
+		return fmt.Errorf("failed to create kinesis client")
+	}
+	return nil
+}
+
+func (k *KinesisSource) GetMetrics() []prometheus.Collector {
+	return nil
+}
+func (k *KinesisSource) GetAggregMetrics() []prometheus.Collector {
+	return nil
+}
+
+func (k *KinesisSource) Configure(yamlConfig []byte, logger *log.Entry) error {
+	config := KinesisConfiguration{}
+	k.logger = logger
```

Comment:
set loglevel

Reply:
The log level is set by the acquisition when it creates the logger.